### PR TITLE
Feature/authentication : the logout

### PR DESCRIPTION
**Pull request by JanStaszewiczEPFL**
### The logout

This PR introduces the logout functionality into the app, along with several related updates and improvements to streamline the implementation and testing process.

**Details**

**Logout Functionality:**

Created a `User.kt` file inside the authentication folder to keep the authentication-related logic clean and organized.
Implemented a logout function in `User.kt` that:

1. Logs the user out.
2. Resets Google Sign-In options.
3. Navigates the user back to the sign-in screen.

Connected this function to the logout button in the `Settings.kt` file.

**Navigation Tests:**

Added navigation tests for the `Settings.kt` file to validate the logout functionality.
However, due to a conflict with a commented version of `Settings.kt` (pushed and merged by @MaxCastelCast), I modified the tests to work with my implementation but left them commented out as agreed upon with the team even though I don't feel like it is the right way to proceed. Then I merged with the new `NavigationTest.kt` file but I couldn't implement my tests are the way of testing the navigation completely changed. 

**Test Cleanup:**

Removed unnecessary tests in `FirebaseAuthTest.kt` to reduce redundancy and improve test suite maintainability.

As for now, we find unnecessary (also discussed and agreed with the coaches) to test the google functions as there is no straight forward way to do so.

### CODE REVIEW

File: app/src/main/java/com/github/se/icebreakrr/authentication/User.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  val googleSignInClient: GoogleSignInClient = GoogleSignIn.getClient(context, gso)
+
+  // Sign out from Firebase
+  auth.signOut()
+
+  // Sign out from Google and reset the google account choice
+  googleSignInClient.signOut().addOnCompleteListener { signOutTask ->
+    if (signOutTask.isSuccessful) {
+      googleSignInClient.revokeAccess().addOnCompleteListener { revokeTask ->
```
Comment (MaxCastelCast) [inline app/src/main/java/com/github/se/icebreakrr/authentication/User.kt:46]: Here the SonarCloud says that the unused function parameter is a major maintanability issue, so it should be removed and replaced with _ , we ll then need to proceed everywhere else in our code where we have unused lambda variables and remove them

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/Settings.kt [hunk 2]
```diff
@@ -89,7 +93,12 @@ fun SettingsScreen(profilesViewModel: ProfilesViewModel, navigationActions: Navi
...
               onClick = {
-                Toast.makeText(context, "Log Out (Not yet implemented)", Toast.LENGTH_SHORT).show()
+                if (isTesting) {
+                  navigationActions.navigateTo(Screen.AUTH)
+                } else {
+                  auth = FirebaseAuth.getInstance()
+                  auth.currentUser?.let { logout(context, navigationActions) }
+                }
               },
               colors = ButtonDefaults.buttonColors(containerColor = Color(0xFFCE0E00)),
```
Comment (MaxCastelCast) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/Settings.kt:103]: Hardcoded color, should be classified in Color.kt and use the constant from there for more maintanability

**Review by MaxCastelCast — CHANGES_REQUESTED** [review]
The code is globally correct, but i have requested some changes that were highlighted by the SonarCloud and some constants issues, alro i wanted to ask if you could let me push py code before yours since i have re-written the tests for NavAction and EndToEnd1 and they work localy, maybe it would be better for you to push with those test already in your code, sorry for the inconvenience

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/Settings.kt (hunk 2 above)
Comment (JanStaszewiczEPFL) [reply app/src/main/java/com/github/se/icebreakrr/ui/sections/Settings.kt:103]: I didn't work on Settings.kt but I can fix that sure

File: app/src/main/java/com/github/se/icebreakrr/authentication/User.kt (hunk 1 above)
Comment (JanStaszewiczEPFL) [reply app/src/main/java/com/github/se/icebreakrr/authentication/User.kt]: You are right, I will change that!

**Review by MaxCastelCast — APPROVED** [review]
Looks good to me !

**Pull request merged**